mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -177,35 +182,36 @@ def dialog_renderer(self):
         # Fall back to main language
         return self.dialog_renderers.get(self._core_lang)
 
+    @property
+    def settings_path(self):
+        return join(xdg.BaseDirectory.save_config_path(BASE_FOLDER, 'skills', self.skill_id), 'settings.json')
+
+    @property
+    def settings_write_path(self):
+        # hopefully nothing in the outside world ever noticed this was not underscore prefixed
+        LOG.warning("self.settings_write_path has been deprecated! do not use it, it is an internal variable")
+        return xdg.BaseDirectory.save_config_path(BASE_FOLDER, 'skills', self.skill_id)
+
     def _init_settings(self):
         """Setup skill settings."""
-
-        # TODO remove this ugly ugly kludge
-        # To not break existing setups,
-        # save to skill directory if the file exists already
-        self.settings_write_path = Path(self.root_dir)
-
-        # Otherwise save to XDG_CONFIG_DIR
-        if not self.settings_write_path.joinpath('settings.json').exists():
-            self.settings_write_path = Path(xdg.BaseDirectory.save_config_path(
-                BASE_FOLDER, 'skills', basename(self.root_dir)))
-
-        # To not break existing setups,
-        # read from skill directory if the settings file exists there
-        settings_read_path = Path(self.root_dir)
-
-        # Then, check XDG_CONFIG_DIR
-        if not settings_read_path.joinpath('settings.json').exists():
-            for path in xdg.BaseDirectory.load_config_paths(BASE_FOLDER, 'skills',
-                                              basename(self.root_dir)):
-                path = Path(path)
-                # If there is a settings file here, use it
-                if path.joinpath('settings.json').exists():
-                    settings_read_path = path
-                    break
-
-        self.settings = get_local_settings(settings_read_path, self.name)
-        self._initial_settings = deepcopy(self.settings)
+        if self.skill_id:  # might not have been set yet when called in __init__ instead of bind
```

Comment:
comment from #9 by @NeonDaniel 
Consider logging a warning or error if this is called in `init`
```python
if not self.skill_id:
    LOG.warning("_init_settings called before skill init completed")
    return
...
```